synapse/storage/background_updates.py

```diff
@@ -772,6 +778,11 @@ def create_index_psql(conn: Connection) -> None:
                 logger.debug("[SQL] %s", sql)
                 c.execute(sql)
 
+                # mypy ignore - `statement_timeout` is defined on PostgresEngine
+                default_timeout = self.db_pool.engine.statement_timeout  # type: ignore[attr-defined]
+                undo_timeout_sql = f"SET statement_timeout = {default_timeout}"
+                c.execute(undo_timeout_sql)
```

Comment:
This needs to be in `finally` section, so if happens even if index creation raises an exception